server.rb

```diff
@@ -35,60 +35,95 @@
     }.to_json
   end
 
-  instance_url = instances.items.first.instance
-  split_url = instance_url.split('/')
-  project, zone, instance_id = split_url[6], split_url[8], split_url[10]
+  instance_data = JSON.parse(instance)
+  puts "returning instance #{instance_data['name']}, formerly in group #{group_name}"
+  content_type :json
+  {
+    name: instance_data['name'],
+    ip:   instance_data['ip']
+  }.to_json
+end
 
-  begin
-    instance = compute.get_instance(project, zone, instance_id)
-  rescue Google::Apis::ClientError
-    # This should probably never happen, unless our url parsing went SUPER wonky
-    content_type :json
-    status 418
-    return {
-      error: 'cannot get specified instance from pool'
-    }.to_json
+Thread.new do
+  loop do
+    config['pools'].each do |pool|
+      if redis.llen(pool['group_name']) < pool['target_size']
+        # TODO: eventually have this support more than just GCE
+        increase_size(group_name, target_size)
+        sleep 30 # Some amount of sleep to avoid race conditions
+      end
+    end
   end
+end
+
+def increase_size(group_name, target_size)
+  size_difference = target_size - redis.llen(group_name)
+  size_difference.each do
+    image_name = group_name.split('-')[3..-1].join('-')
```

Comment:
Maybe pass in `payload` from the request, which has the `image_name` in it (as well as machine type!)

Reply:
Ooh, that would make sense, but since we split off this size increase into a background loop and not as part of the request, I don't think we have the payload here?

Reply:
🤦‍♂️ right. we should probably have it as part of the config then?